mempool/v0/clist_mempool.go

```diff
@@ -534,7 +534,8 @@ func (mem *CListMempool) ReapMaxBytesMaxGas(maxBytes, maxGas int64) types.Txs {
 
 		txs = append(txs, memTx.tx)
 
-		dataSize := types.ComputeProtoSizeForTxs([]types.Tx{memTx.tx})
+		// we subtract 4 here to account for extra bytes we add to the additional data fields
+		dataSize := types.ComputeProtoSizeForTxs([]types.Tx{memTx.tx}) - 4
```

Comment:
[trying to understand] If we "add extra bytes", why do we subtract from the space taken by a transaction? Don't we want to do the opposite?

Reply:
I was similarly confused by these modification. Fortunately it was reverted in https://github.com/celestiaorg/celestia-core/pull/897 (a PR to celestia-core after the audit commit hash)

Reply:
FWIW evidence was previously added to data so it may have alluded to that

Reply:
If you folks have removed the "- 4" from the latest version, then we can close this comment